hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/impl/TokenCreatePrecompile.java

```diff
@@ -569,210 +455,68 @@ public Bytes getFailureResultFor(final ResponseCodeEnum status) {
-        return getTokenCreateWrapperFungible(tokenCreateStruct, isFungible, initSupply, decimals, aliasResolver);
-    }
-
-    public static TokenCreateWrapper decodeTokenCreateWithoutFeesV2(
-            @NonNull final Tuple tokenCreateStruct,
-            final boolean isFungible,
-            final BigInteger initSupply,
-            final BigInteger decimals,
-            final UnaryOperator<byte[]> aliasResolver) {
-        return getTokenCreateWrapperFungible(tokenCreateStruct, isFungible, initSupply, decimals, aliasResolver);
-    }
-
-    private static TokenCreateWrapper getTokenCreateWrapperFungible(
-            @NonNull final Tuple tokenCreateStruct,
-            final boolean isFungible,
-            final BigInteger initSupply,
-            final BigInteger decimals,
-            final UnaryOperator<byte[]> aliasResolver) {
-        final var tokenName = (String) tokenCreateStruct.get(0);
-        final var tokenSymbol = (String) tokenCreateStruct.get(1);
-        final var tokenTreasury = convertLeftPaddedAddressToAccountId(tokenCreateStruct.get(2), aliasResolver);
-        final var memo = (String) tokenCreateStruct.get(3);
-        final var isSupplyTypeFinite = (Boolean) tokenCreateStruct.get(4);
-        final var maxSupply = (long) tokenCreateStruct.get(5);
-        final var isFreezeDefault = (Boolean) tokenCreateStruct.get(6);
-        final var tokenKeys = decodeTokenKeys(tokenCreateStruct.get(7), aliasResolver);
-        final var tokenExpiry = decodeTokenExpiry(tokenCreateStruct.get(8), aliasResolver);
+        final Tuple decodedArguments = decodeFunctionCall(
+                input, CREATE_FUNGIBLE_TOKEN_METHOD_V1.selector, CREATE_FUNGIBLE_TOKEN_METHOD_V1.decoder);
 
-        return new TokenCreateWrapper(
-                isFungible,
-                tokenName,
-                tokenSymbol,
-                tokenTreasury.getAccountNum() != 0 ? tokenTreasury : null,
-                memo,
-                isSupplyTypeFinite,
-                initSupply,
-                decimals,
-                maxSupply,
-                isFreezeDefault,
-                tokenKeys,
-                tokenExpiry);
+        return getTokenCreateWrapperFungible(
+                decodedArguments.get(0), decodedArguments.get(1), decodedArguments.get(2), aliasResolver);
     }
 
     /**
      * Decodes the given bytes of the fungible token.
      *
-     * <p><b>Important: </b>This is an old version of this method and is superseded by
-     * decodeFungibleCreateWithFeesV2(). The selector for this function is derived from:
-     * createFungibleTokenWithCustomFees((string,string,address,string,bool,uint32,bool,(uint256,(bool,address,bytes,
-     * bytes,address))[],(uint32,address,uint32)),uint256,uint256,(uint32,address,bool,bool,address)[],
-     * (uint32,uint32,uint32,uint32,bool,address)[])
+     * <p><b>Important: </b>This is the latest version and supersedes decodeFungibleCreate(). The
```

Comment:
I guess this comment is correct but should there be a similar one for `decodeFungibleCreateV3`?